ims/einstein/ceph.py

```diff
@@ -132,11 +162,45 @@ def remove(self, img_id):
 
     @log
     def write(self, img_id, data, offset):
+        """
+        Write data to the image
+
+        :param img_id: what the image is called
+        :param data: the data to be written
+        :param offset: where to start writing data
+        :return: the number of bytes written
+        """
+        try:
+            with self.__open_image(img_id) as img:
+                size = img.size()
+                if(offset + len(data) < size):
+                    bytes_written = img.write(data, offset)
+                    return bytes_written
+        except rbd.ImageNotFound:
+            raise file_system_exceptions.ImageNotFoundException(img_id)
+        except rbd.InvalidArgument:
+            raise file_system_exceptions.ArgumentsOutOfRangeException()
+
+    @log
+    def read(self, img_id, offset, length):
+        """
+        Read data from the image
+
+        :param img_id: what the image is called
+        :param offset: the offset to start reading at
+        :param length: how many bytes to read
+        :return: a string of the data read
+        """
         try:
             with self.__open_image(img_id) as img:
-                img.write(data, offset)
+                size = img.size()
+                if(offset + length < size):
```

Comment:
@xukan that is good, but I guess that line 198 will raise the exception and the if statement will not allow line 198 to execute when it is violated. So no exception will be raised ?